bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java

```diff
@@ -1116,6 +1144,40 @@ private void setThingStatus(Thing thing, ThingStatusInfo thingStatusInfo) {
         }
     }
 
+    @Override
+    public void setEnabled(ThingUID thingUID, boolean enabled) {
+        Thing thing = getThing(thingUID);
+
+        if (enabled) {
+            // Enable a thing
+            logger.debug("Thing {} will be enabled.", thing.getUID().getAsString());
+            if (storage != null) {
+                storage.remove(thingUID.getAsString());
+            }
+            if (!thing.getStatus().equals(ThingStatus.ONLINE)) {
+                setThingStatus(thing, buildStatusInfo(ThingStatus.UNINITIALIZED, ThingStatusDetail.NONE));
```

Comment:
This look like pretty broad big-hammer-strategy to me. What happens if e.g.
* the thing was e.g. in `REMOVING` before?
* the thing was e.g. in `OFFLINE`/`COMMUNICATION_ERROR` before? 
* somebody calls `setEnabled(..., true) on an already enabled thing?

Reply:
!thing.getStatus().equals(ThingStatus.ONLINE) changed to !ThingHandlerHelper.isHandlerInitialized(thing) as suggested, this will prevent the attempt to initialize already initialized thing. Otherwise will change the storage entry and try to initialize the thing.